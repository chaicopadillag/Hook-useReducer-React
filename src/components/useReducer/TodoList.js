import React from 'react';
import PropTypes from 'prop-types';
import TodoListItem from './TodoListItem';

const TodoList = ({ todos, handleToogle, handleDelete }) => {
	return (
		<ul className="list-group list-group-flush">
			{todos.map((todo, index) => {
				return <TodoListItem todo={todo} index={index} handleToogle={handleToogle} handleDelete={handleDelete} key={todo.id} />;
			})}
		</ul>
	);
};
TodoList.propTypes = {
	todos: PropTypes.array.isRequired,
	handleToogle: PropTypes.func.isRequired,
	handleDelete: PropTypes.func.isRequired,
};

export default TodoList;
