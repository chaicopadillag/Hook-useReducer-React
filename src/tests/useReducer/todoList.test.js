import { shallow } from 'enzyme';
import TodoList from '../../components/useReducer/TodoList';
import demoTodos from '../demoTodos';

describe('Prueba de Todo List', () => {
	const handleDelete = jest.fn(),
		handleToggle = jest.fn();
	let wrapper = shallow(<TodoList todos={demoTodos} index={1} handleDelete={handleDelete} handleToogle={handleToggle} />);

	test('debe de mostrar correctamente', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('debe tener dos <TodoListItem/>', () => {
		expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
		expect(wrapper.find('TodoListItem').at(0).prop('handleDelete')).toEqual(expect.any(Function));
	});
});
