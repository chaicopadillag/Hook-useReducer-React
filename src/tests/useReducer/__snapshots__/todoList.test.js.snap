// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de Todo List debe de mostrar correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToogle={[MockFunction]}
    index={0}
    key="1"
    todo={
      Object {
        "estado": false,
        "id": 1,
        "tarea": "Aprender Mongo",
      }
    }
  />
  <TodoListItem
    handleDelete={[MockFunction]}
    handleToogle={[MockFunction]}
    index={1}
    key="2"
    todo={
      Object {
        "estado": false,
        "id": 2,
        "tarea": "Aprender Firebase",
      }
    }
  />
</ul>
`;
